web/client/utils/SecurityUtils.js

```diff
@@ -184,18 +184,22 @@ const SecurityUtils = {
             return parameters;
         }
     },
-    addAuthenticationToSLD: function(layerOptions, options) {
-        if (layerOptions.SLD) {
-            const parsed = URL.parse(layerOptions.SLD, true);
-            const params = SecurityUtils.addAuthenticationParameter(layerOptions.SLD, parsed.query, options.securityToken);
-            return assign({}, layerOptions, {
+    addAuthenticationToSLD: function(layerParams, options) {
+        // excluding null values for CQL_FILTER that are causing point 3_
+        // here https://github.com/geosolutions-it/MapStore2/issues/4674#issuecomment-580300446
+        const {CQL_FILTER, ...layerPrms} = layerParams;
```

Comment:
> Generally speaking seems correct. Please fix as in the comments.
> 
> * [ ]  Wasn't able to test widgets on map because of #4853 please make sure that everything is still compatible

nothing has changed for widgets in map

> * [ ]  `cqlAlreadyAnded` is a little ugly. If confirmed that is necessary, we can use a more significative name.

yes is necessary, i tried to  change a bit the name

> * [ ]  `availableDependenciesForEditingWidgetSelector` should reuse part of the code, make only sure that the information added to this are not needed in the use cases of the other one.

i think they are quite different so cannot take in input the availableDependenciesSelector, not sure about how to reuse something here.

Reply:
> - If it is not enough, in `Legend.jsx` you can exclude parameters that can cause errors (invalid CQL_FILTER) from query object in a specific function

i will do this